### PR TITLE
Compressor updates

### PR DESCRIPTION
**Pull request by chicm-ms**
1. Change optimizer to an optional parameter for pruner
2. Fix an pruner example bug

### CODE REVIEW

File: src/sdk/pynni/nni/compression/torch/activation_rank_filter_pruners.py [hunk 1]
```diff
@@ -177,7 +179,7 @@ class ActivationMeanRankFilterPruner(ActivationRankFilterPruner):
     https://arxiv.org/abs/1611.06440
     """
 
-    def __init__(self, model, config_list, optimizer, activation='relu', statistics_batch_num=1):
+    def __init__(self, model, config_list, optimizer=None, activation='relu', statistics_batch_num=1):
```
Comment (QuanluZhang) [inline src/sdk/pynni/nni/compression/torch/activation_rank_filter_pruners.py]: APoZ should also be updated

File: src/sdk/pynni/nni/compression/torch/pruners.py [hunk 2]
```diff
@@ -70,14 +72,16 @@ class AGP_Pruner(Pruner):
     https://arxiv.org/pdf/1710.01878.pdf
     """
 
-    def __init__(self, model, config_list, optimizer):
+    def __init__(self, model, config_list, optimizer=None):
```
Comment (QuanluZhang) [inline src/sdk/pynni/nni/compression/torch/pruners.py]: optimizer cannot be None in AGP pruner because it is an iterative pruner

Comment (QuanluZhang) [reply src/sdk/pynni/nni/compression/torch/pruners.py]: or give default value None to the argument optimizer, and check optimizer in `__init__`, if it is None raise error with message "AGP pruner is an iterative pruner, please pass optimizer of the model to it"

File: src/sdk/pynni/nni/compression/torch/weight_rank_filter_pruners.py [hunk 3]
```diff
@@ -178,7 +184,7 @@ class FPGMPruner(WeightRankFilterPruner):
     https://arxiv.org/pdf/1811.00250.pdf
     """
 
-    def __init__(self, model, config_list, optimizer):
+    def __init__(self, model, config_list, optimizer=None):
```
Comment (QuanluZhang) [inline src/sdk/pynni/nni/compression/torch/weight_rank_filter_pruners.py]: FPGM is also an iterative pruner, optimizer should not be None

File: src/sdk/pynni/nni/compression/torch/activation_rank_filter_pruners.py (hunk 1 above)
Comment (chicm-ms) [reply src/sdk/pynni/nni/compression/torch/activation_rank_filter_pruners.py]: Thanks, fixed.

File: src/sdk/pynni/nni/compression/torch/pruners.py (hunk 2 above)
Comment (chicm-ms) [reply src/sdk/pynni/nni/compression/torch/pruners.py]: Thanks, fixed.

File: src/sdk/pynni/nni/compression/torch/weight_rank_filter_pruners.py (hunk 3 above)
Comment (chicm-ms) [reply src/sdk/pynni/nni/compression/torch/weight_rank_filter_pruners.py]: Thanks, fixed.

File: examples/model_compress/model_prune_torch.py [hunk 4]
```diff
@@ -179,6 +179,8 @@ def test(model, device, test_loader):
 
 def main(args):
     device = torch.device('cuda') if torch.cuda.is_available() else torch.device('cpu')
+    if not os.path.exists(args.checkpoints_dir):
+        os.makedirs(args.checkpoints_dir)
```
[the remark this answers is not in the document]
Comment (chicm-ms) [reply examples/model_compress/model_prune_torch.py]: thanks, updated.

**Pull request merged**